### PR TITLE
Large files

### PR DESCRIPTION
**Pull request by MarcelHoogAntink**
Solved issue #71 'snapshot:load fails with DB >100Mb'
tested with sql file >800 Mb, also test with gz file.

### CODE REVIEW

File: src/Bigfile.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class Bigfile
+{
+    protected $file;
+
+    public function __construct($filename, $mode = 'r', $compressed = false)
+    {
+        if (! file_exists($filename)) {
+            throw new \Exception('File not found');
+        }
+        if ($compressed) {
```
Comment (freekmurze) [inline src/Bigfile.php]: Use a ternary so we can lose that `else` statement.

File: src/Bigfile.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+namespace Spatie\DbSnapshots;
+
+class Bigfile
+{
+    protected $file;
+
+    public function __construct($filename, $mode = 'r', $compressed = false)
+    {
+        if (! file_exists($filename)) {
+            throw new \Exception('File not found');
```
Comment (freekmurze) [inline src/Bigfile.php]: Import this so we can get rid of that trailing `\`. Also do this for the other classes.

File: src/Bigfile.php (hunk 1 above)
Comment (MarcelHoogAntink) [reply src/Bigfile.php]: changed

File: src/Bigfile.php (hunk 2 above)
Comment (MarcelHoogAntink) [reply src/Bigfile.php]: changed

**Comment by freekmurze** [comment]
Could you add a test?

**Comment by MarcelHoogAntink** [comment]
Sure, but how to detect the correct working as it depends on the envirionment (amout of available memory). So the existing tests already are doing the job?

**Comment by bilfeldt** [comment]
@MarcelHoogAntink might I ask why you closed your PR?

A way to test this could be to set a low memory limit in PHPunit and then seed a database which we know is larger than the specified memory limit.

Setting the memory limit to only 16M:
```xml
<php>
  <ini name="memory_limit" value="16M" />
</php>
```

Method to get the memory limit in bytes:
```php
/*
 * Converts a number with byte unit (B / K / M / G) into an integer.
 *
 * @see https://stackoverflow.com/questions/10208698/checking-memory-limit-in-php
 */
function unitToInt($s)
{
    return (int)preg_replace_callback('/(\-?\d+)(.?)/', function ($m) {
        return $m[1] * pow(1024, strpos('BKMG', $m[2]));
    }, strtoupper($s));
}
```

Test checking that the dump filesize is actually larger than the memory limit:
```php
// $snapshot referes to Spatie\DbSnapshots\Snapshot

$mem_limit = $this->unitToInt(ini_get('memory_limit'));
$filesize = $snapshot->disk->size($snapshot->fileName); // See: https://laravel.com/api/5.8/Illuminate/Filesystem/FilesystemAdapter.html#method_size

$this->assertGreaterThan($mem_limit, $filesize);
```

**Pull request closed without merging**